### ISSUE TITLE
Add support to map ruler based on configured projection

### ISSUE THREAD

**Issue by barcelosleo** [python-visualization/folium#1963]
It would be great if we had the possibility to show a ruler around the map. This ruler should change as the projection change as well, and the frequency of the ticks should be configurable as well (eg. show the latitude every 10km and longitude every 5km if we are dealing with an UTM projection or latitude every 2º and longitude every 3º if we are dealing with geographic projection). It should be something like that:
![image](https://github.com/python-visualization/folium/assets/7603208/329562b3-cf6e-4fb7-8d54-9546fd0492ab)

We already have a scale bar, which is great, but it would be very good to be able to see the ruler around the map.

**Comment by hansthen** [comment]
Do you mean something like this? https://github.com/dtutic/Leaflet.EdgeScaleBar

**Comment by barcelosleo** [comment]
@hansthen yes, but instead of scale I would like to plot the coordinates. The second print I posted translates very well what I need, where the UTM coordinates are shown.

**Comment by Conengmo** [comment]
You could see if Leaflet has a plugin that supports this. If not, maybe some custom Javascript has to be written to make this, perhaps based off of https://github.com/dtutic/Leaflet.EdgeScaleBar.

In any case, we'll have to see if it's something we incorporate in Folium. We try to not take on too much additional maintenance. I can see it as something you may host yourself, and we link to it in our docs. Alternatively, take a look at out [plugin acceptance criteria](https://github.com/python-visualization/folium/blob/main/.github/CONTRIBUTING.md#plugin-acceptance-criteria).
